shell/browser/api/electron_api_web_contents.cc

```diff
@@ -2572,6 +2619,43 @@ std::vector<content::NavigationEntry*> WebContents::GetHistory() const {
   return history;
 }
 
+void WebContents::RestoreHistory(
+    v8::Isolate* isolate,
+    gin_helper::ErrorThrower thrower,
+    int index,
+    const std::vector<v8::Local<v8::Value>>& entries) {
+  if (!web_contents()
+           ->GetController()
+           .GetLastCommittedEntry()
+           ->IsInitialEntry()) {
+    thrower.ThrowError(
+        "Cannot restore history on webContents that have previously loaded "
+        "a page.");
+    return;
+  }
+
+  auto navigation_entries = std::make_unique<
+      std::vector<std::unique_ptr<content::NavigationEntry>>>();
+
+  for (const auto& entry : entries) {
+    content::NavigationEntry* nav_entry = nullptr;
+    if (!gin::Converter<content::NavigationEntry*>::FromV8(isolate, entry,
+                                                           &nav_entry) ||
+        !nav_entry) {
+      // Invalid entry, bail out early
+      return;
```

Comment:
I think this should throw so applications will know the history was not restored.

Reply:
You got it, added!

This is _currently_ never going to happen, so I can't test it - Chromium will handle garbage pageState and attempt to restore the rest of the navigation entry (https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/navigation_controller_impl_browsertest.cc;l=12454-12464)